src/Launcher.vala

```diff
@@ -42,17 +42,17 @@ public class Dock.Launcher : Gtk.Button {
 
                 var context = Gdk.Display.get_default ().get_app_launch_context ();
                 context.set_timestamp (Gdk.CURRENT_TIME);
+                context.launched.connect (() => {
```

Comment:
We also need to connect to `launch_failed` so that we don't end up leaving the animation if the launch doesn't succeed for some reason